src/ark/segmentation/fiber_segmentation.py

```diff
@@ -294,9 +353,147 @@ def segment_fibers(data_xr, fiber_channel, out_dir, fov, blur=2, contrast_scalin
+        fov_length = fov_fiber_img.shape[0]
+        fov_table = fiber_object_table[fiber_object_table.fov == fov]
+
+        # fov level stats
+        fov_align_scores = fov_table['alignment_score'].values
+        fov_alignment.append(np.mean(fov_align_scores[~np.isnan(fov_align_scores)]))
+        fov_avg_length.append(np.mean(fov_table['major_axis_length'].values))
+        fov_p_density, fov_p_density = calculate_density(fov_table, fov_length**2)
+        fov_pixel_density.append(fov_p_density)
+        fov_fiber_density.append(fov_p_density)
+
+        # tile level stats
+        t_alignment, t_length, t_pixel_density, t_fiber_density = [], [], [], []
+        fov_list, tile_x, tile_y = [], [], []
+
+        # create tiles based on provided tile_length
+        for i in range(int(fov_length / tile_length)):
+            y_range = (i*tile_length, (i + 1) * tile_length)
+            for j in range(int(fov_length / tile_length)):
+                x_range = (j * tile_length, (j + 1) * tile_length)
+                fov_list.append(fov)
+                tile_x.append(x_range[0])
+                tile_y.append(y_range[0])
+
+                tile_fiber_img = fov_fiber_img[y_range[0]:y_range[1], x_range[0]:x_range[1]]
+                tile_fiber_img[tile_fiber_img > 0] = 1
+                if save_tiles:
+                    if not os.path.exists(os.path.join(save_dir, fov)):
+                        os.makedirs(os.path.join(save_dir, fov))
+                    io.imsave(os.path.join(save_dir, fov, f'tile_{y_range[0]},{x_range[0]}.tiff'),
+                              tile_fiber_img, check_contrast=False)
+
+                tile_table = fov_table[np.logical_and(
+                    fov_table['centroid-0'] >= y_range[0], fov_table['centroid-0'] < y_range[1])]
+                tile_table = tile_table[np.logical_and(
+                    tile_table['centroid-1'] >= x_range[0], tile_table['centroid-1'] < x_range[1])]
+
+                if len(tile_table) < 5:
+                    for stat_list in [t_alignment, t_length, t_pixel_density, t_fiber_density]:
+                        stat_list.append(np.nan)
+                else:
+                    # alignment stat
+                    align_scores = tile_table['alignment_score'].values
+                    align_scores = align_scores[~np.isnan(align_scores)]
+                    avg_alignment = np.mean(align_scores) if len(align_scores) >= 5 else np.nan
+                    t_alignment.append(avg_alignment)
+
+                    # length stat
+                    avg_length = np.mean(tile_table['major_axis_length'].values)
+                    t_length.append(avg_length)
+
+                    # density stats
+                    pixel_density, fiber_density = calculate_density(tile_table, tile_length**2)
+                    t_pixel_density.append(pixel_density)
+                    t_fiber_density.append(fiber_density)
```

Comment:
There is a lot going on here in this loop. One suggestion would be grab these lines and place them into a helper function which gets called.